lib/ukstore/include/uk/store.h

```diff
@@ -29,3 +29,345 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
+
+#ifndef __UK_STORE_H__
+#define __UK_STORE_H__
+
+#include <string.h>
+#include <uk/config.h>
+#include <uk/assert.h>
+#include <uk/list.h>
+#include <uk/arch/atomic.h>
+#include <uk/essentials.h>
+#include <uk/alloc.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !CONFIG_LIBUKSTORE
+#define uk_store_libid(libname) (-1)
+
+enum uk_store_entry_type;
+struct uk_store_entry;
+
+#define __UK_STORE_STATIC_ENTRY(entry, lib_str, e_type, e_get, e_set)	\
+	static const struct uk_store_entry				\
+	__unused __align8						\
+	__uk_store_entries_list ## _ ## entry = {			\
+		.name = STRINGIFY(entry),				\
+		.type       = (UK_STORE_ENTRY_TYPE(e_type)),		\
+		.get.e_type = (e_get),					\
+		.set.e_type = (e_set),					\
+		.flags      = UK_STORE_ENTRY_FLAG_STATIC		\
+	}
+
+/* Do not call directly */
+#define _UK_STORE_STATIC_ENTRY(entry, lib_str, e_type, e_get, e_set)	\
+	__UK_STORE_STATIC_ENTRY(entry, lib_str, e_type, e_get, e_set)
+
+/**
+ * Adds an entry to the entry section of a library.
+ *
+ * @param entry the entry in the section
+ * @param e_type the type of the function
+ * @param e_get getter pointer
+ * @param e_set setter pointer
+ */
+#define UK_STORE_STATIC_ENTRY(entry, e_type, e_get, e_set)	\
+	_UK_STORE_STATIC_ENTRY(entry, STRINGIFY(__LIBNAME__),	\
+		e_type, e_get, e_set)
+
+/* Flags if an entry is static or dynamic */
+#define UK_STORE_ENTRY_FLAG_STATIC	(1 << 0)
+
+#define UK_STORE_ENTRY_ISSTATIC(entry)	\
+	((entry)->flags & UK_STORE_ENTRY_FLAG_STATIC)
+
+#define uk_store_get_entry(libname, foldername, entryname)	\
+	do { } while (0)
```

Comment:
But, actually, the more I think about this, we should not provide any replacements for query calls. If someone wants to retrieve or set something from/to `ukstore`, that code has to make sure that the dependency to `lib/ukstore` is fulfilled. It is just convenience if we can transparently comment out with the preprocessor entry registration from libraries as soon as `ukstore` is not part of a built.

Reply:
Ok, I'm deleting everything except the entry registration